### ISSUE TITLE
getAllHotspotDevices - Performance optimization

### ISSUE THREAD

**Issue by webmaster-imaginet** [hypery2k/cordova-hotspot-plugin#29]
I have performance issues with **getHotspotDevices**

To show this, I change a little bit your sample app controller.

**Here is the code:**
###### 

.controller('DevicesCtrl', function ($scope, $interval) {
    "use strict";
    var init = function () {

```
$scope.data = {};
$scope.data.clockTime = 0;
$scope.data.clock = $interval(function(){
    $scope.data.clockTime ++;
    console.log('%c%s','font-size:14px;color:green;',"Sec: " + $scope.data.clockTime);
}, 1000);

  console.log('%c%s','font-size:20px;color:red;',"START: " + $scope.data.clockTime);
      cordova.plugins.hotspot.getAllHotspotDevices(
        function (response) {
          console.table(response);
          $scope.devices = response;
            console.log('%c%s','font-size:20px;color:red;',"STOP: " + $scope.data.clockTime);
            $interval.cancel($scope.data.clock);
        }, function () {
          alert('Device listing failed.');
        }
      );
};

// init controllers
//init();

// API
$scope.refresh = function () {
  init()
};
```
###### 

**it took 108 second to get the devices list**

![image](https://cloud.githubusercontent.com/assets/4920443/14640354/d698b062-0649-11e6-8096-1d99a4b4d795.png)

I filmed a video to show you the problems.
https://www.dropbox.com/s/msofovh7w58u7ft/hotspot_plugin_slow.mp4?dl=0

**Comment by hypery2k** [comment]
please try if the current master is acceptable. Could not really reduce it more due to technically restrictions

**Comment by webmaster-imaginet** [comment]
I used version 1.1.2 (cordova plugin add cordova-plugin-hotspot)
I also try version 1.1.1 (cordova plugin add cordova-plugin-hotspot!1.1.1) but got the same "slow" results.

now I try to use the git "master" as you asked (cordova plugin add https://github.com/hypery2k/cordova-hotspot-plugin.git) but got empty array

![image](https://cloud.githubusercontent.com/assets/4920443/14641181/add7f684-064d-11e6-8a88-5de8aed0f51b.png)

**Comment by hypery2k** [comment]
Should be better with 1.1.3

**Comment by webmaster-imaginet** [comment]
Much faster...

![image](https://cloud.githubusercontent.com/assets/4920443/14642423/116943a6-0653-11e6-8e89-16e862f4737c.png)

**But why reading the arp file is not working super fast?**

Is it will be faster to read the file and send as callBack the result "AS IS" and then do the divide to IP & MAC in the javascript?
If the answer is yes, maybe is good idea to add a new methood for that?

**Comment by hypery2k** [comment]
the ARP my not contain all devices. But I optimize the method more

**Comment by webmaster-imaginet** [comment]
Good!
This is still ver 1.1.3 (from the NPM) or will be 1.1.4?

**Comment by hypery2k** [comment]
will be 1.1.4
But I'll wait if any other issue comes up before releasing next version

**Comment by hypery2k** [comment]
Released version 1.1.4
